rca/landingpages/models.py

```diff
@@ -560,9 +562,61 @@ class Meta:
         verbose_name = "Landing Page - Enterprise"
 
 
+class EELandingPageRelatedEditorialPage(RelatedPage):
+    source_page = ParentalKey(
+        "landingpages.EELandingPage", related_name="related_editorial_pages"
+    )
+    panels = [PageChooserPanel("page", ["editorial.EditorialPage"])]
+
+
 class EELandingPage(BasePage):
     template = "patterns/pages/editorial_event_landing/editorial_event_landing.html"
     max_count = 1
 
+    news_link_text = models.TextField(
+        max_length=120,
+        blank=False,
+        null=True,
+        help_text=_("The text do display for the link"),
+    )
+    news_link_target_url = models.URLField(blank=False, null=True)
+
     class Meta:
         verbose_name = "Landing Page - Editorial and Events"
+
+    @property
+    def news(self):
```

Comment:
Maybe either make this a cached property or make it a method and call in get page context (I think I favour the latter these days, I read somewhere (vague) its best not have DB queries hidden behind properties)

Reply:
Swapped for a method and called in get_context